### ISSUE TITLE
DeterministicSourcePaths doesn't take NuGet source packages into account

### ISSUE THREAD

**Issue by clairernovotny** [dotnet/roslyn#42995]
**Version Used**: 5.0p1

**Steps to Reproduce**:

1. Create a class library
2. Add a package reference to xunit.assert.source
3. enable `Deterministic` and `ContinuousIntegrationBuild`

**Expected Behavior**:
A source root should be created and the path to the source normalized. 

**Actual Behavior**:
The source paths for the source package point to the NuGet directory and are not normalized.

Possible solution:
NuGet has a variable for its package root. We can add a SourceRoot for that and that'd cover all source packages under that root:

```xml
  <ItemGroup>
    <SourceRoot Include="$(NuGetPackageRoot)" Condition="'$(NuGetPackageRoot)' != ''" />
  </ItemGroup>
```

**Comment by tmat** [comment]
Need to think about where this should be handled. I don't think Roslyn is the right place since the compiler should not depend on (or assume existence of) a specific package manager, like NuGet.

**Comment by clairernovotny** [comment]
@tmat fair enough. perhaps it should be part of the NuGet targets as that's what turns the source package items into Compile items at restore time?
